config/application.rb

```diff
@@ -101,9 +101,6 @@ class Application < Rails::Application
     # Reconsider whether to enable again after upgrading to Rails 7.1, where the size of the header is limited to 1KB
     config.action_view.preload_links_header = false
 
-    # Make Ruby preserve the timezone of the receiver when calling `to_time`.
-    config.active_support.to_time_preserves_timezone = false
```

Comment:
This seems to be an actual change that I can't grasp the full impact of. I wonder if we should apply this change only in a separate PR because of the possible complication when doing time zone dependent operations.

Presently I see the following:
```ruby
[3] pry(main)> "2016-04-23T10:23:12.000Z".to_time
=> 2016-04-23 13:23:12 +0300
```

With the change it would be:
```ruby
[5] pry(main)> "2016-04-23T10:23:12.000Z".to_time
=> 2016-04-23 10:23:12 +0000
```

So my suggestion is to revert this (although it is deprecated but should still be working) and apply it once we see stable operation with Rails 7.1

Reply:
The message is just a warning that we are relying on a thing that is going to be removed in the next major version. It is not an error. So we don't know whether we are affected until some time operations returns an unexpected time zone.

We have a lot of billing logic relying on Time objects (not Time.zone) so I'd prefer that we keep the deprecated configuration and apply it separately, probably put some effort into investigating a little bit more about it's possible effects. As well apply it without other significant changes to more easily detect issues or changes in behavior with it.